### ISSUE TITLE
Externalize Event-detection API to benefit other projects

### ISSUE THREAD

**Issue by d-karl** [RLBot/RLBot#73]
Hey guys,

this is an enhancement request. From what I can see, you have a powerful injector that can detect events inside Rocket League and make the game state available to outside programs. This state is, according to your wiki: https://github.com/RLBot/RLBot/wiki/Input-and-Output-Data-(current)#values-from-game. I assume this data is kept current.

Over at https://github.com/d-karl/Rocket-League-Chroma-Control, I have dormant code that was used to display Razer Chroma (RGB lighting) effects based on events happening within the game, such as goal scored, countdown etc. Example video of when it worked is here: https://www.youtube.com/watch?v=fZzJ_VMt-6U (countdown "go" effect is not final and was change later). 

Long story short: event detection was broken when Psyonix adjusted the Unreal Engine logging output I was using to detect events. The data available via your framework would be stupifyingly useful for me, in that I could even detect boosting, which side scored a goal and much more.

Is there any way you guys could externalize this and make it available to me, or would I be allowed to use your injector (or code relating to it) for my purposes? What's the best way to do this? It would be neat if the injector could be built as a library, with an API for retrieving the current game state. I don't currently know if that is possible? Thanks guys.

**Comment by dtracers** [comment]
Hi,

Do to the way our framework works we can only detect events when offline in exhibition mode.  So I feel for your purposes this is now much more limited.  And not nearly as useful.

**Comment by d-karl** [comment]
I see, thanks for letting me know. I assume this is by design, because you don't want to interfere with online matches at all. Have you ever tried to also detect events in online matches and found it to be impossible or did you simply not investigate the possibility since it wasn't relevant to your use case? Would I in theory be able to extend the functionality to online matches?

**Comment by tarehart** [comment]
Yep, by design. I suspect that detecting events online would be doable. 

Part of our system is closed source, and we grant access pretty cautiously. Also keep in mind that our strategy breaks with some RL updates too. (But so far our expert has always been able to adapt).

I'll talk to the team about this.

**Comment by ddthj** [comment]
Due to our integration with the psyonix API, we can only pull information from the game when it is locked in an "offline mode." This is no longer a doable enhancement.
